presto-hive/src/test/java/com/facebook/presto/hive/authentication/TestKerberosRules.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.authentication;
+
+import com.facebook.presto.hive.HdfsConfiguration;
+import com.facebook.presto.hive.HdfsConfigurationUpdater;
+import com.facebook.presto.hive.HiveClientConfig;
+import com.facebook.presto.hive.HiveHdfsConfiguration;
+import com.google.common.collect.ImmutableSet;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import sun.security.krb5.Config;
+
+import javax.security.auth.Subject;
+import javax.security.auth.kerberos.KerberosPrincipal;
+
+import static org.apache.hadoop.security.UserGroupInformationShim.createUserGroupInformationForSubject;
+import static org.apache.hadoop.security.authentication.util.KerberosName.resetDefaultRealm;
+import static org.testng.Assert.assertEquals;
+import static java.util.Collections.emptySet;
+
+public class TestKerberosRules
+{
+    private String originalConf;
+
+    @BeforeClass
+    public void setup()
+    {
+        originalConf = System.getProperty("java.security.krb5.conf");
+
+        // default realm in this file is REALM1.COM
+        System.setProperty("java.security.krb5.conf", "src/test/resources/mock-krb5.conf");
```

Comment:
Doing something like this is very unreliable, as it may interfere with the other tests.

Reply:
@arhimondr I see. I was not aware that the tests are executed in parallel. I will remove the modification of system properties from this place.